### PR TITLE
Robustify cloud deployment + include initial KEDA configuration

### PR DESCRIPTION
**Pull request by pablodanswer**
## Description

- Ensure cloud tags point to private Dockerhub images to prevent confusion for open source users
- Initial KEDA configurations (limits are set very conservatively, but I am actively iterating on them- best to simply get the base configuration in the repo for maintainability)

### CODE REVIEW

File: backend/danswer/background/celery/apps/app_base.py [hunk 1]
```diff
@@ -146,6 +146,8 @@ def on_celeryd_init(sender: Any = None, conf: Any = None, **kwargs: Any) -> None
 def wait_for_redis(sender: Any, **kwargs: Any) -> None:
     """Waits for redis to become ready subject to a hardcoded timeout.
     Will raise WorkerShutdown to kill the celery worker if the timeout is reached."""
+    if MULTI_TENANT:
```
Comment (rkuo-danswer) [inline backend/danswer/background/celery/apps/app_base.py]: Understand we don't need the checks for multi-tenant/cloud, but conditionals like this should be as high in the call stack as possible, not embedded in tool functions.

File: .github/workflows/docker-build-push-backend-container-on-tag.yml [hunk 2]
```diff
@@ -3,61 +3,61 @@ name: Build and Push Backend Image on Tag
 on:
   push:
     tags:
-      - '*'
+      - "*"
 
 env:
-  REGISTRY_IMAGE: danswer/danswer-backend
+  REGISTRY_IMAGE: ${{ contains(github.ref_name, 'cloud') && 'danswer/danswer-backend-cloud' || 'danswer/danswer-backend' }}
   LATEST_TAG: ${{ contains(github.ref_name, 'latest') }}
```
Comment (rkuo-danswer) [inline .github/workflows/docker-build-push-backend-container-on-tag.yml:10]: is this going to potentially tag cloud images as latest?  should that be explicitly wired off if so?

**Review by rkuo-danswer — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-build-push-backend-container-on-tag.yml (hunk 2 above)
Comment (pablonyx) [reply .github/workflows/docker-build-push-backend-container-on-tag.yml:10]: Believe we do want this functionality within docker– the latest tag should apply separately to the docker repositories (ie open source users would never see the `latest` cloud tags, is my understanding)

File: backend/danswer/background/celery/apps/app_base.py (hunk 1 above)
Comment (pablonyx) [reply backend/danswer/background/celery/apps/app_base.py]: Completely agree! Moved all these checks up to the worker init

File: .github/workflows/docker-build-push-backend-container-on-tag.yml (hunk 2 above)
Comment (rkuo-danswer) [reply .github/workflows/docker-build-push-backend-container-on-tag.yml:10]: ok, completely separate docker images. sounds good!

**Review by rkuo-danswer — APPROVED** [review]

**Pull request merged**